server/src/main/java/org/opensearch/search/approximate/ApproximatePointRangeQuery.java

```diff
@@ -0,0 +1,591 @@
+
+                        @Override
+                        public long cost() {
+                            return reader.maxDoc();
+                        }
+                    };
+                } else {
+                    if (sortOrder.equals(SortOrder.ASC)) {
+                        return new ScorerSupplier() {
+
+                            final DocIdSetBuilder result = new DocIdSetBuilder(reader.maxDoc(), values, pointRangeQuery.getField());
+                            final PointValues.IntersectVisitor visitor = getIntersectVisitor(result);
+                            long cost = -1;
+
+                            @Override
+                            public Scorer get(long leadCost) throws IOException {
+                                intersectLeft(values.getPointTree(), visitor, size);
+                                DocIdSetIterator iterator = result.build().iterator();
+                                return new ConstantScoreScorer(weight, score(), scoreMode, iterator);
+                            }
+
+                            @Override
+                            public long cost() {
+                                if (cost == -1) {
+                                    // Computing the cost may be expensive, so only do it if necessary
+                                    cost = values.estimateDocCount(visitor);
+                                    assert cost >= 0;
+                                }
+                                return cost;
+                            }
+                        };
+                    }
+                    return new ScorerSupplier() {
+
+                        final DocIdSetBuilder result = new DocIdSetBuilder(reader.maxDoc(), values, pointRangeQuery.getField());
+                        final PointValues.IntersectVisitor visitor = getIntersectVisitor(result);
+                        long cost = -1;
+
+                        @Override
+                        public Scorer get(long leadCost) throws IOException {
+                            intersectRight(values.getPointTree(), visitor, size);
+                            DocIdSetIterator iterator = result.build().iterator();
+                            return new ConstantScoreScorer(weight, score(), scoreMode, iterator);
+                        }
+
+                        @Override
+                        public long cost() {
+                            if (cost == -1) {
+                                // Computing the cost may be expensive, so only do it if necessary
+                                cost = values.estimateDocCount(visitor);
+                                assert cost >= 0;
+                            }
+                            return cost;
+                        }
+                    };
+                }
+            }
+
+            @Override
+            public Scorer scorer(LeafReaderContext context) throws IOException {
+                return pointRangeQueryWeight.scorer(context);
```

Comment:
shouldn't this be invoking your scorerSupplier.get?

Reply:
whoops, fixed